### PR TITLE
Use move in gripper_action when max_effort is 0

### PR DESCRIPTION
**Pull request by marcbone**
This PR implements the [my suggested workaround ](https://github.com/frankaemika/franka_ros/pull/173#issuecomment-964994857) from #173.

This PR solves #172. 

Now the gripper_action can close the gripper to a smaller width when max_effort is 0. However, since this is a breaking change we want to wait until the following issue is resolved in moveit (@rickstaa):

- [x]   [#2956 MoveIt: Make max_effort settable in the MoveIt Simple Controller Manager](https://github.com/ros-planning/moveit/issues/2956)

### CODE REVIEW

**Comment by rickstaa** [comment]
@marcbone Thanks for implementing this. I am currently still trying to fix the gravity compensator (i.e. #160) but I will jump right on it after that is done.

**Comment by rickstaa** [comment]
@marcbone I implemented the requested functionality in Moveit (see https://github.com/ros-planning/moveit/pull/2984). It is currently under review. I will let you know when it is merged.

**Comment by rickstaa** [comment]
@marcbone I tested your pull request, and I think your solution is very intuitive. It works perfectly when the gripper hand is positioned horizontally. Please be aware that the move action is currently broken for non-horizontal gripper hand orientations (see https://github.com/frankaemika/franka_ros/pull/173#issuecomment-979223037 and https://github.com/frankaemika/franka_ros/pull/173#issuecomment-979249735).

**Review by rickstaa — APPROVED** [review]

**Comment by rickstaa** [comment]
@marcbone Just a heads-up. https://github.com/ros-planning/moveit/pull/2984#event-5769897890 has been merged.

**Comment by rickstaa** [comment]
@marcbone, @gollth Just wondering, do you need me to implement something else on the MoveIt side before this can be merged?

Also, did you already manage to find good PID gains that can control the franka_gripper in the vertical mode that does not break the velocity checking logic in the `grasp` server (see https://github.com/frankaemika/franka_ros/pull/173#issuecomment-979223037)?

**Comment by gollth** [comment]
> Also, did you already manage to find good PID gains that can control the franka_gripper in the vertical mode that does not break the velocity checking logic in the `grasp` server (see [#173 (comment)](https://github.com/frankaemika/franka_ros/pull/173#issuecomment-979223037))?

Not yet unfortunately but we have it on the radar

**Comment by rickstaa** [comment]
Great! Thanks a lot!

**Comment by marcbone** [comment]
> Just wondering, do you need me to implement something else on the MoveIt side before this can be merged?

No, basically I am just waiting for the moveit release, since I guess nobody is going to build moveit from source.

**Comment by rickstaa** [comment]
@marcbone Ha good point! I overlooked that fact. Let's wait for the release then.

**Comment by rickstaa** [comment]
@marcbone Just a heads up, I tried to rebase this branch on the `develop` branch but git runs into conflicts of which two are not flagged by git.  The following lines are injected by the git merge tool on line `354-355` of the `franka_gripper_sim.cpp` file without git flagging them:

```cpp
double width = this->finger1_.getPosition() + this->finger2_.getPosition();  // recalculate
bool ok = goal->width - eps.inner < width and width < goal->width + eps.outer;
```

This will throw a error, further the following line will be changed from:

```cpp
result.success = static_cast<decltype(result.success)>(true);
```

to

```cpp
result.success = static_cast<decltype(result.success)>(ok);
```

Which also throws an error. After fixing these lines and solving the other flagged conflicts, the pull request works again (see https://github.com/rickstaa/franka_ros/tree/use-move-in-gripper-action-rebased).

**Pull request merged**
